spi/src/main/java/org/opensearch/ml/common/spi/tools/Tool.java

```diff
@@ -127,5 +129,7 @@ interface Factory<T extends Tool> {
          * @return the default tool version
          */
         String getDefaultVersion();
+
+        List<String> getAllModelKeys();
```

Comment:
Since most of the tools will return an empty list, can you declare a default function?

Reply:
Otherwise, when a new Tool added, we cannot make sure all related model ID fields are detected. And that may skip the logic checking agent before deleting the model.

Reply:
I think it's like an attribute of Tool like version/name. Do you want to implement another class to default return the empty list in this function?